### ISSUE TITLE
Inconsistent color contrast calculation

### ISSUE THREAD

**Issue by DhirajKhot** [dequelabs/axe-core#2931]
Product: axe-core 

Expectation: The axe-core engine should calculate the correct color contrast for the same styled element consistently.

Actual: The color contrast ratio differs everytime upon running the scan and sometimes reports the violated result for the same styled element.

Motivation: The result should be consistent for the same styled element

<pre><code>
axe-core version: 4.2.0
We are using the cypress-axe wrapper to run the axe engine as we have the integration tests written in cypress.
</code></pre>

Below is the snapshot of the computed style of the element for which issue is being reported - 
![image](https://user-images.githubusercontent.com/20378231/117962901-62638b80-b33d-11eb-8ef0-12642d1311d5.png)

For the same element we get violations with different color contrast ratios - 
![image](https://user-images.githubusercontent.com/20378231/117963562-2aa91380-b33e-11eb-9a0b-b232ddceefaa.png)
![image](https://user-images.githubusercontent.com/20378231/117963494-15cc8000-b33e-11eb-81dd-c2d814486498.png)
![image](https://user-images.githubusercontent.com/20378231/117963602-3a285c80-b33e-11eb-8ab4-d76e76cee2eb.png)

I tried logging the actual value of the color of the desired element during the run. It logs the correct desired color value - 
![image](https://user-images.githubusercontent.com/20378231/117965316-401f3d00-b340-11eb-9b66-08df6ee102b6.png)

Color logged is rgb(96, 94, 92) which is #605e5c
But the axe is assigning fgColor as #9b9998 and hence violating the rule for color contrast.

Any idea what could be the issue here?

**Comment by straker** [comment]
Interesting. Would it be possible to publish the failing test to something public like a gist or codepen? Color contrast should always return the same results so long as the element in question is static, that is isn't part of something that uses JavaScript / CSS animations to change state.

**Comment by DhirajKhot** [comment]
@straker Thanks for looking into this. 
I will take a look if I can do that, but seems difficult to publish and repro the issue on a public platform. 

Anything else if I can help with for more data?

**Comment by straker** [comment]
Unfortunately color-contrast bugs require me being able to debug the code as it's running against the problem as it's a very complex part of the code. Without that I'm not sure what we can do.

**Comment by straker** [comment]
Going to close this issue. Please reopen if you are still running into this and can provide a test case that reproduces the issue.
